### PR TITLE
Various minor code cleanups found by scan-build

### PR DESCRIPTION
**Pull request by LordAro**
Various things found by running `scan-build sh -c './configure && make'` with all checkers turned on. Mostly missing null checks and dead code. Vast numbers of false positives that were ignored.

Documentation fixes were found by going through the output of `doxygen`, ignoring missing documentation warnings. Nearly all fixed, but some left that aren't clear how to be fixed (usually related to attributes being referred to before they're declared? Maybe the explicit `#` could be removed)

Commits are almost entirely self-contained, so probably best to review individually

### CODE REVIEW

File: src/town_cmd.cpp [hunk 1]
```diff
@@ -377,8 +377,7 @@ void Town::UpdateVirtCoord()
 	SetDParam(0, this->index);
 	SetDParam(1, this->cache.population);
 	this->cache.sign.UpdatePosition(pt.x, pt.y - 24 * ZOOM_LVL_BASE,
-		_settings_client.gui.population_in_label ? STR_VIEWPORT_TOWN_POP : STR_VIEWPORT_TOWN,
-		STR_VIEWPORT_TOWN);
+		_settings_client.gui.population_in_label ? STR_VIEWPORT_TOWN_POP : STR_VIEWPORT_TOWN);
```
Comment (frosch123) [inline src/town_cmd.cpp]: This breaks ViewportAddTownNames.

File: src/viewport.cpp [hunk 2]
```diff
@@ -1313,9 +1313,8 @@ static void ViewportAddSigns(DrawPixelInfo *dpi)
  * @param center the (preferred) center of the viewport sign
  * @param top    the new top of the sign
  * @param str    the string to show in the sign
- * @param str_small the string to show when zoomed out. STR_NULL means same as \a str
  */
-void ViewportSign::UpdatePosition(int center, int top, StringID str, StringID str_small)
+void ViewportSign::UpdatePosition(int center, int top, StringID str)
```
Comment (frosch123) [inline src/viewport.cpp]: This breaks ViewportAddTownNames.

File: src/viewport.cpp [hunk 3]
```diff
@@ -1328,9 +1327,6 @@ void ViewportSign::UpdatePosition(int center, int top, StringID str, StringID st
 	this->center = center;
 
 	/* zoomed out version */
-	if (str_small != STR_NULL) {
```
Comment (frosch123) [inline src/viewport.cpp]: This breaks ViewportAddTownNames.

File: src/newgrf_spritegroup.cpp [hunk 4]
```diff
@@ -154,6 +154,7 @@ static uint32 RotateRight(uint32 val, uint32 rot)
 {
 	/* Do not rotate more than necessary */
 	rot %= 32;
+	assert(rot > 0);
```
Comment (frosch123) [inline src/newgrf_spritegroup.cpp:157]: Is "== 0" valid?

File: src/window.cpp [hunk 5]
```diff
@@ -2752,18 +2752,17 @@ static void HandleAutoscroll()
 	y -= vp->top;
 
 	/* here allows scrolling in both x and y axis */
-#define scrollspeed 3
+	const int SCROLLSPEED = 3;
```
Comment (frosch123) [inline src/window.cpp]: static const?

File: Doxyfile [hunk 6]
```diff
@@ -130,7 +130,7 @@ RECURSIVE              = YES
 EXCLUDE                =
 EXCLUDE_SYMLINKS       = NO
 EXCLUDE_PATTERNS       = */3rdparty \
-                         */.svn \
+                         */.git \
```
Comment (frosch123) [inline Doxyfile]: .git is in the root folder, not in the src folder.

File: src/cargopacket.cpp [hunk 7]
```diff
@@ -602,7 +602,7 @@ uint VehicleCargoList::Reassign<VehicleCargoList::MTA_DELIVER, VehicleCargoList:
  * Returns reserved cargo to the station and removes it from the cache.
  * @param max_move Maximum amount of cargo to move.
  * @param dest Station the cargo is returned to.
- * @param ID of next the station the cargo wants to go next.
+ * @param next ID of next the station the cargo wants to go next.
```
Comment (frosch123) [inline src/cargopacket.cpp]: There are too many "next" in the sentence :)

File: src/station_base.h [hunk 8]
```diff
@@ -515,8 +515,7 @@ class AirportTileIterator : public OrthogonalTileIterator {
 
 public:
 	/**
-	 * Construct the iterator.
-	 * @param ta Area, i.e. begin point and width/height of to-be-iterated area.
+	 Construct the iterator.
```
Comment (frosch123) [inline src/station_base.h]: Missing *

**Comment by nielsmh** [comment]
The Doxygen config changes should probably be moved to a separate PR from the code cleanup.

File: src/viewport.cpp (hunk 3 above)
Comment (LordAro) [reply src/viewport.cpp]: That's annoying, it does. Some part of this was definitely flagged up as a redundant assignment, and I could've sworn I checked it properly...

File: src/newgrf_spritegroup.cpp (hunk 4 above)
Comment (LordAro) [reply src/newgrf_spritegroup.cpp:157]: The point it that it shouldn't be, as then you're doing `val << 32`, which isn't defined all that well. If we want to just assume that it should be 0, that's a bit different.

That said, not sure why this exists at all, when ROR in bitmath_func.hpp exists.

**Review by frosch123 — APPROVED** [review]

**Pull request merged**